source/exe/hot_restart.h

```diff
@@ -201,5 +205,39 @@ class HotRestartImpl : public HotRestart,
   bool parent_terminated_{};
 };
 
+#else  // __linux__
+
+/**
+ * No-op implementation of HotRestart for everybody else.
```

Comment:
On this point, can we potentially just revert all the changes to hot_restart_impl.h/cc, and just add a new hot_restart_nop_impl.h/cc, and do select() compile? This will keep existing code cleaner and make it possible to compile out hot restart support. (I think MSFT people will want this too initially).